api/v1beta1/osccluster_validation.go

```diff
@@ -61,6 +61,16 @@ func ValidateCidr(cidr string) (string, error) {
 	return cidr, nil
 }
 
+// ValidateSubregionName check that subregionName is a valid az format
+func ValidateSubregionName(subregionName string) (string, error) {
+	switch {
+	case strings.HasSuffix(subregionName, "1a") || strings.HasSuffix(subregionName, "1b") || strings.HasSuffix(subregionName, "2a") || strings.HasSuffix(subregionName, "2b"):
```

Comment:
Maybe use a regexp ?

https://github.com/outscale/cluster-api-provider-outscale/blob/main/api/v1beta1/oscmachine_validation.go#L205

Reply:
I just copied the bit of code from the vm validation code which had the same function. But yeah we could clean this up a bit.